docs/model_developers/modules.rst

```diff
@@ -122,3 +122,96 @@ The format for requires in requirements.txt is the folllowing:
  * Or a repository location such as  git+https://github.com/project/python-foo The correct syntax
    to use is then: eggname@git+https://../repository#branch with branch being optional.
 
+
+Working on modules
+==================
+Since modules often have dependencies on other modules, it is common to develop against multiple
+modules (or a project and one or more modules) simultaneously. One might for example need to
+extend a dependent module to add support for some new feature. Because this use case is so common,
+this section will describe how to work on multiple modules simultaneously so that any changes are
+visible to the compiler. This procedure is of course applicable for working on a single module as well.
+
+Setting up the dev environment
+------------------------------
+To set up the development environment for a project, activate your development Python environment and
+install the project with ``inmanta project install``. To set up the environment for a single module,
+run ``inmanta module install`` instead.
+
+The following subsections explain any additional steps you need to take if you want to make changes
+to one of the dependency modules as well.
+
+v1 modules
+^^^^^^^^^^
+Any modules you find in the project's ``modulepath`` after starting from a clean project and setting
+up the development environment are v1 modules. You can make changes to these modules and they will
+be reflected in the next compile. No additional steps are required.
+
+v2 modules
+^^^^^^^^^^
+All other modules are v2 and have been installed by ``inmanta project install`` into the active Python
+environment. If you want to be able to make changes to one of these modules, the easiest way is to
+check out the module repo separately and run ``inmanta module install`` on it, overwriting the published
+package that was installed previously This will install the module in editable form: any changes you make
+to the checked out files will be picked up by the compiler. You can also do this prior to installing the
```

Comment:
```suggestion
check out the module repo separately and run ``inmanta module install`` on it, overwriting the published
package that was installed previously. This will install the module in editable form: any changes you make
to the checked out files will be picked up by the compiler. You can also do this prior to installing the
```